Cesium3DTilesContent/src/LegacyUtilities.cpp

```diff
@@ -0,0 +1,181 @@
+#include <Cesium3DTilesContent/LegacyUtilities.h>
+#include <CesiumAsync/IAssetResponse.h>
+#include <CesiumGltf/Accessor.h>
+#include <CesiumGltf/Buffer.h>
+#include <CesiumGltf/BufferView.h>
+#include <CesiumGltf/Model.h>
+#include <CesiumGltfContent/GltfUtilities.h>
+#include <CesiumUtility/Uri.h>
+
+#include <glm/glm.hpp>
+
+#include <algorithm>
+#include <functional>
+#include <iterator>
+#include <string>
+
+namespace Cesium3DTilesContent {
+namespace LegacyUtilities {
+using namespace CesiumGltf;
+
+std::optional<uint32_t> parseOffset(
+    const rapidjson::Document& document,
+    const char* semantic,
+    CesiumUtility::ErrorList& errorList) {
+  const auto semanticIt = document.FindMember(semantic);
+  if (semanticIt == document.MemberEnd() || !semanticIt->value.IsObject()) {
+    return {};
+  }
+  const auto byteOffsetIt = semanticIt->value.FindMember("byteOffset");
+  if (byteOffsetIt == semanticIt->value.MemberEnd() ||
+      !isValue<uint32_t>(byteOffsetIt->value)) {
+    errorList.emplaceError(
+        std::string("Error parsing feature table, ") + semantic +
+        "does not have valid byteOffset.");
+    return {};
+  }
+  return getValue<uint32_t>(byteOffsetIt->value);
+}
+
+bool validateJsonArrayValues(
+    const rapidjson::Value& arrayValue,
+    uint32_t expectedLength,
+    ValuePredicate predicate) {
+  if (!arrayValue.IsArray()) {
+    return false;
+  }
+
+  if (arrayValue.Size() != expectedLength) {
+    return false;
+  }
+
+  for (rapidjson::SizeType i = 0; i < expectedLength; i++) {
+    if (!std::invoke(predicate, arrayValue[i])) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+std::optional<glm::dvec3>
+parseArrayValueDVec3(const rapidjson::Value& arrayValue) {
+  if (validateJsonArrayValues(arrayValue, 3, &rapidjson::Value::IsNumber)) {
+    return std::make_optional(glm::dvec3(
+        arrayValue[0].GetDouble(),
+        arrayValue[1].GetDouble(),
+        arrayValue[2].GetDouble()));
+  }
+  return {};
```

Comment:
This probably resolves to a `std::nullopt`, but do you mind writing it out for a slow and easily-confused coworker like me? 😃 

```suggestion
  return std::nullopt;
```

Reply:
I personally use "return {};" as a kind of 21st century "return NULL;". It works in a lot of different contexts as a null value: smart pointers, std::optional, even pointers though I do use nullptr for that. I'll gladly follow the Cesium native coding conventions, but {} is concise and, I think, pretty useful.

Reply:
That's fair, and I'm not sure if this goes against an actual item in the coding guide. It's a personal preference, but feel free to skip